lib/src/command/barback.dart

```
@@ -2,231 +2,23 @@
-
-    sourceDirectories.addAll(dirs);
-  }
-
-  /// Adds the default sources that should be used if no directories are passed
-  /// on the command line.
-  void _addDefaultSources() {
-    sourceDirectories.addAll(defaultSourceDirectories
-        .where((dir) => dirExists(entrypoint.root.path(dir))));
-
-    // TODO(rnystrom): Hackish. Assumes there will only be one or two
-    // default sources. That's true for pub build and serve, but isn't as
-    // general as it could be.
-    if (sourceDirectories.isEmpty) {
-      var defaults;
-      if (defaultSourceDirectories.length == 1) {
-        defaults = 'a "${defaultSourceDirectories.first}" directory';
-      } else {
-        defaults = '"${defaultSourceDirectories[0]}" and/or '
-            '"${defaultSourceDirectories[1]}" directories';
-      }
-
-      dataError("Your package must have $defaults,\n"
-          "or you must specify the source directories.");
-    }
-  }
-
-  /// Converts a list of [directoryNames] to a sentence.
-  ///
-  /// After the list of directories, [singularVerb] will be used if there is
-  /// only one directory and [pluralVerb] will be used if there are more than
-  /// one. Then [suffix] is added to the end of the sentence, and, finally, a
-  /// period is added.
-  String _directorySentence(Iterable<String> directoryNames,
-      String singularVerb, String pluralVerb, String suffix) {
-    var directories =
-        pluralize('Directory', directoryNames.length, plural: 'Directories');
-    var names = toSentence(directoryNames.map((dir) => '"$dir"'));
-    var verb =
-        pluralize(singularVerb, directoryNames.length, plural: pluralVerb);
-
-    var result = "$directories $names $verb";
-    if (suffix != null) result += " $suffix";
-    result += ".";
-
-    return result;
+    log.error(log.red("Dart 2 has a new build system. Learn how to migrate "
+        "from ${log.bold('pub build')} and\n"
+        "${log.bold('pub serve')}: https://webdev.dartlang.org/dart-2\n"));
```

Nit: use `fail()` instead of `log.error()` and `exit()`.

---

Done